### PR TITLE
Fix package import and add travis

### PR DESCRIPTION
**Pull request by nth10sd**
(Upgrade pip and setuptools in your virtualenv as well)
```
pushd ~/autobisect/ && ~/debug-autobisect-bug/thirdpy35/bin/pip install --upgrade pip setuptools && ~/debug-autobisect-bug/thirdpy35/bin/pip install --upgrade -r requirements.txt && popd
```

The only caveat is that one has to be in the `autobisect` repo directory while installing via `requirements.txt`.

```
$ ~/debug-autobisect-bug/thirdpy35/bin/python
Python 3.5.2 (default, Nov 23 2017, 16:37:01) 
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import ffpuppet
>>> dir(ffpuppet)
['BrowserTerminatedError', 'BrowserTimeoutError', 'FFPuppet', 'LaunchError', '__all__', '__author__', '__builtins__', '__cached__', '__doc__', '__file__', '__loader__', '__name__', '__package__', '__path__', '__spec__', 'core', 'helpers', 'minidump_parser', 'puppet_logger', 'workers']
>>> import autobisect
>>> dir(autobisect)
['__builtins__', '__cached__', '__doc__', '__file__', '__loader__', '__name__', '__package__', '__path__', '__spec__', 'absolute_import', 'bisect', 'build_manager', 'builds', 'config', 'evaluator', 'main']
>>>
```

As you can see, one can access `'bisect', 'build_manager', 'builds', 'config', 'evaluator', 'main'` in the `autobisect` package.

Develop mode still installs correctly even with these changes:
```
$ ~/debug-autobisect-bug/thirdpy35/bin/pip install -e .
Obtaining file:///home/winworklin/autobisect
/snip
Installing collected packages: autobisect
  Running setup.py develop for autobisect
Successfully installed autobisect
```

Without these changes:

```
$ ~/debug-autobisect-bug/thirdpy35/bin/pip uninstall autobisect
/snip
  Successfully uninstalled autobisect-0.0.1

$ ~/debug-autobisect-bug/thirdpy35/bin/pip install --upgrade .
Processing /home/winworklin/autobisect
/snip
Installing collected packages: autobisect
  Running setup.py install for autobisect ... done
Successfully installed autobisect-0.0.1

$ ~/debug-autobisect-bug/thirdpy35/bin/python
Python 3.5.2 (default, Nov 23 2017, 16:37:01) 
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import ffpuppet
>>> dir(ffpuppet)
['BrowserTerminatedError', 'BrowserTimeoutError', 'FFPuppet', 'LaunchError', '__all__', '__author__', '__builtins__', '__cached__', '__doc__', '__file__', '__loader__', '__name__', '__package__', '__path__', '__spec__', 'core', 'helpers', 'minidump_parser', 'puppet_logger', 'workers']
>>> import autobisect
>>> dir(autobisect)
['__builtins__', '__cached__', '__doc__', '__file__', '__loader__', '__name__', '__package__', '__path__', '__spec__']
>>>
```

Without these changes, one cannot access `'bisect', 'build_manager', 'builds', 'config', 'evaluator', 'main'` in the `autobisect` package.

In both cases, `ffpuppet` works correctly, and I also cross-checked with `funfuzz`.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,4 +1,3 @@
-configparser>=3.5.0
-fuzzfetch==0.5.7
--e git://github.com/MozillaSecurity/lithium@master#egg=lithium-reducer
```
Comment (pyoor) [inline requirements.txt]: The #egg format is specifically required in order to use pipenv.  Unless you see a reason not to use them, I think we should revert this change.

File: requirements.txt [hunk 2]
```diff
@@ -1,4 +1,3 @@
-configparser>=3.5.0
-fuzzfetch==0.5.7
```
Comment (pyoor) [inline requirements.txt]: Why remove these dependencies?

File: tox.ini [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    # This contains builds of flake8 that we don't want to check
+    dist/*
+max-line-length = 120
+show-source = true
+statistics = true
+
+docstring-quotes = """
+multiline-quotes = """
+
+[isort]
```
Comment (pyoor) [inline tox.ini]: Isn't this supposed to be pylint?

File: tox.ini [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+force_single_line=true
+force_sort_within_sections=true
+line_length=119
+
+[pytest]
+addopts = --durations=10 -v --cache-clear
```
Comment (pyoor) [inline tox.ini]: Should we be running flake8 and pylint here?

File: requirements.txt (hunk 1 above)
Comment (nth10sd) [reply requirements.txt]: Ok, I'll revert this portion. However, please ensure this still works for you using pipenv.

File: requirements.txt (hunk 2 above)
Comment (nth10sd) [reply requirements.txt]: They weren't removed - they now live in `setup.py`, and if you notice the `.` in requirements.txt, when you do a `pip install --user --upgrade -r requirements.txt`, the command will also install `configparser` and `fuzzfetch`.

File: setup.py [hunk 5]
```diff
@@ -8,29 +8,46 @@
...
+            'Programming Language :: Python :: 3.5',
+            'Programming Language :: Python :: 3.6',
+            'Programming Language :: Python :: 3.7',
         ],
         description='Automatic bisection utility for Mozilla Firefox and SpiderMonkey',
         entry_points={
-            "console_scripts": ["autobisect = autobisect.main:main"]
+            'console_scripts': ['autobisect = autobisect.main:main'],
         },
-        install_requires=["configparser", "ffpuppet", "fuzzfetch", "lithium-reducer"],
```
[the remark this answers is not in the document]
Comment (nth10sd) [reply setup.py]: They are in `requirements.txt`, however, I can reinstate this.

File: tox.ini (hunk 3 above)
Comment (nth10sd) [reply tox.ini]: They are. They work with `flake8-commas`, `flake8-isort` and `flake8-quotes` together with `flake8`.

File: tox.ini (hunk 4 above)
Comment (nth10sd) [reply tox.ini]: We could, if that's preferred. I'll make this change.

Comment (nth10sd) [reply tox.ini]: Also, the only linter here is flake8. pylint seems like it will be separate work.

File: requirements.txt (hunk 2 above)
Comment (nth10sd) [reply requirements.txt]: Let me know if you prefer to have them back in, e.g. for pipenv or some of the other 100000 ways of dealing with pip ;-)

File: setup.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (pyoor) [reply setup.py]: I was under the impression that both needs to exist.  Requirements.txt for the application requirements (version matched) and setup.py for library requirements.

File: requirements.txt (hunk 2 above)
Comment (pyoor) [reply requirements.txt]: I was under the impression that both needs to exist. Requirements.txt for the application requirements (version matched) and setup.py for library requirements.

File: setup.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (nth10sd) [reply setup.py]: This was already reinstated.
